src/linking/remove_internal_symbols.cpp

```diff
@@ -148,6 +169,22 @@ void remove_internal_symbols(
     }
   }
 
+  std::vector<irep_idt> diff;
+  std::set_difference(
```

Comment:
I'd use `log.conditional_output(...)` for this entire block here as there is no need to compute a `set_difference` if warnings wouldn't even get printed.